php/class-ee-db.php

```diff
@@ -2,325 +2,482 @@
 			return false;
 		}
+
+		$set_keys     = array_keys( $values );
+		$set_bindings   = array_values( $values );
+		$where_bindings = $this->where['bindings'];
+
+		$set_clause = implode( $set_keys, ' = ?, ' ) . ' = ?';
+
+		$query         = "UPDATE $this->tables SET $set_clause{$this->where['query_string']}";
+		$pdo_statement = self::$pdo->query( $query );
+
+		$counter = 0;  //We need counter here as we need to bind values of both  SET and WHERE clauses
+
+		foreach ( $set_bindings as $binding ) {
+			$pdo_statement->bindValue( ++ $counter, $binding );
+		}
+
+		foreach ( $where_bindings as $binding ) {
+			$pdo_statement->bindValue( ++ $counter, $binding );
+		}
+
+		$result = $pdo_statement->execute();
+
+		if ( ! $result ) {
+			EE::debug( self::$pdo->errorInfo() );
+
+			return false;
+		}
+
+		return true;
 	}
 
 	/**
-	 * Check if a site entry exists in the database as well as if it is enbaled.
+	 * Delete data from table.
 	 *
-	 * @param String $site_name Name of the site to be checked.
+	 * @throws Exception If no table are specified or multiple table are specified or no where clause are specified
 	 *
-	 * @return bool true  if site is enabled,
-	 *              false if disabled or site does not exists.
+	 * @return bool Success.
 	 */
-	public static function site_enabled( $site_name ) {
+	public function delete() {
+		if ( empty( $this->tables ) ) {
+			throw new Exception( 'Delete: No table specified' );
+		}
 
-		if ( empty ( self::$db ) ) {
-			self::init_db();
+		if ( empty( $this->where ) ) {
+			throw new Exception( 'Delete: No where clause specified' );
+		}
+
+		if ( strpos( $this->tables, ',' ) !== false ) {
+			throw new Exception( 'Delete: Multiple table specified' );
 		}
 
-		$site = self::select( array( 'id', 'is_enabled' ), array( 'sitename' => $site_name ) );
+		$query = "DELETE FROM $this->tables{$this->where['query_string']}";
+
+		$pdo_statement = self::$pdo->query( $query );
 
-		if ( 1 === count( $site ) ) {
-			return $site[0]['is_enabled'];
+		foreach ( $this->where['bindings'] as $key => $binding ) {
+			$pdo_statement->bindValue( $key + 1, $binding );
 		}
 
-		return false;
+		$result = $pdo_statement->execute();
+
+		if ( ! $result ) {
+			EE::debug( self::$pdo->errorInfo() );
+
+			return false;
+		}
+
+		return true;
 	}
 
 	/**
-	 * Get site type.
+	 * Returns a query fragment for where clause
+	 *
+	 * If the param given is ['column', 100], it returns ['column = ?', 100]
+	 * If the param given is ['column', '>', 100], it returns ['column > ?', 100]
+	 *
+	 * @param array $condition An array of format [column, operator, value] or [column, value]
 	 *
-	 * @param String $site_name Name of the site.
+	 * @throws Exception
 	 *
-	 * @return string type of site.
+	 * @return array prepared query string and its corresponding binding
 	 */
-	public static function get_site_command( $site_name ) {
+	private function get_where_fragment( $condition ) {
+		$column   = $condition[0];
```

Comment:
Please add one condition before using `$condition[0]` here and return if below condition is not satisfied.

> if ( ! is_array( $condition ) || empty( $condition ) || count( $condition ) > 3 ) {
>  // return or exeption.....  
>}

Reply:
why below, what's the point checking empty after that ??? 
you know that if you try to access index 0 value if the variable is empty it will give you warning/error right? That's why I specifically asked you to add it before this line.

Reply:
Ohh... Okay. I've moved the condition to the top and added `array` type hint in parameter(and removed `is_array` from if condition ).